code/modules/antagonists/revenant/revenant.dm

```diff
@@ -109,6 +109,12 @@
 		revealed = FALSE
 		incorporeal_move = INCORPOREAL_MOVE_JAUNT
 		invisibility = INVISIBILITY_REVENANT
+		if(staticOverlays.len)
+			for(var/mob/living/simple_animal/drone/D in GLOB.drones_list)
+				if(D && D.client && D.seeStatic)
+					for(var/image/I in staticOverlays)
+						D.staticOverlays.Remove(I)
+						D.client.images.Remove(I)
```

Comment:
you can pass a list to Remove, allowing you to offload the inner loop to byond internals.